### ISSUE TITLE
tsne plot different for wine dataset

### ISSUE THREAD

**Issue by MdAsifKhan** [ShelsonCao/DNGR#1]
Hi,
I was trying to replicate your results, but seems to get different performance for wine dataset. Here is 2D tsne plot of 32 representation learned from sae. If you're using any specific parameter configuration please share.
![wine](https://cloud.githubusercontent.com/assets/12439236/18625317/75004ad8-7e56-11e6-988a-36bcc1cd2642.png)

Thanks
Asif

**Comment by ShelsonCao** [comment]
Dear Asif,
The variable of _data_ should be a similarity matrix (178 by 178) in command "DNGR(_data_, sae, opts, nnsize, Kstep, alpha)", not the original feature matrix (178 by 13), which is followed the paper by Tian et al. 2014. How to build such a similarity matrix? As described  in Section 4.1 in our paper, you need to use cosine similarity of two different instances as edge weights, and labels as ground truth.
Thanks!
Shelson

**Comment by MdAsifKhan** [comment]
Yes I am using cosine similarity matrix for wine dataset.

**Comment by ShelsonCao** [comment]
How do you initialize the neural network parameters of wine dataset? I use the following settings: nnsize = [178 128 32];  sae.ae{1}.learningRate = 0.25; sae.ae{2}.learningRate = 0.25; opts.numepochs = 80; opts.batchsize = 1;

**Comment by MdAsifKhan** [comment]
Hi Shelson Cao,
I tried with above parameter configuration still get the same results. Maybe it has to do something with dataset. I downloaded wine dataset from here https://gist.github.com/tijptjik/9408623 . To create graph I computed cosine similarity between samples. To do tsne plot I use 32 D feature vector. 

Have you done any preprocessing with data before computing cosine similarity like: normalization etc. 
Can you please post your example for wine dataset.

**Comment by ShelsonCao** [comment]
I have sent the datasets via email. And I set perplexity=30, initial_dims=32 in t-sne tool.

**Comment by MdAsifKhan** [comment]
Thanks Shelson..!!
It works well with your data. Also I have implemented your work using keras (https://github.com/MdAsifKhan/DNGR-Keras).
